rules/crowdstrike_rules/crowdstrike_dns_request.py

```diff
@@ -8,10 +8,7 @@ def rule(event):
     # We need to run either for Crowdstrike.DnsRequest or for Crowdstrike.FDREvent with the
     # 'DnsRequest' fdr_event_type. Crowdstrike.DnsRequest is covered because of the
     # association with the type
-    if (
-        event.get("p_log_type") == "Crowdstrike.FDREvent"
-        and event.get("fdr_event_type", "") != "DnsRequest"
-    ):
+    if not contains_crowdstrike_fdr_event_type(event, "DnsRequest"):
```

Comment:
Might be confused, but the helper on line 236, will return `False` when `p_log_type == "Crowdstrike.DNSRequest"`, right? `not False` is `True`here so the rule will return early? 🤔

Reply:
😄  
If you mean when `p_log_type` is `"Crowdstrike.DNSRequest"`,
The helper on line 236, will effectively run `"Crowdstrike.DNSRequest" != "Crowdstrike.FDREvent"`, which is true, so it'll enter line 237 and will return `True`.

Reply:
Oh, you're right sorry for the misunderstanding here 😅